### ISSUE TITLE
Local path for header file

### ISSUE THREAD

**Issue by KemalPajevic** [facebook/facebook-ios-sdk#1664]
## Checklist

- [x] I've updated to the latest released version of the SDK
- [x] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [x] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [x] I've read the [Code of Conduct](https://github.com/facebook/facebook-ios-sdk/blob/master/CODE_OF_CONDUCT.md)
- [x] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Xcode Version: `12.4`
- Swift Version: `5.3.2`
- Installation Platform & Verison: `Cocoapods` version `1.10.0`
- FacebookSDK Version: `9.0.1`

## Goals

When hitting a breakpoint, I want to print put some object using the `po` command.

## Expected Results

The description of the object is printed in the console.

## Actual Results

I get the following error message:
```
error: virtual filesystem overlay file '/Users/kylinchang/Library/Developer/Xcode/DerivedData/FacebookSDK-axnectjfufffeccdyyliqwafpkfw/Build/Intermediates.noindex/ArchiveIntermediates/FBSDKCoreKit-Dynamic/IntermediateBuildFilesPath/FBSDKCoreKit.build/Release-iphonesimulator/FBSDKCoreKit-Dynamic.build/all-product-headers.yaml' not found

error: couldn't IRGen expression. Please check the above error messages for possible root causes.
```

There seems to be a local path to @KylinChang's user library.

## Steps to Reproduce

Just enter any breakpoint and try to print out any object.

**Comment by joesus** [comment]
Interesting. I tried to get rid of this error previously by updating our scripts to build the frameworks with the `-no-serialize-debugging-options` flag and it doesn't seem to have prevented the issue.

The most comprehensive article I found on the topic is https://steipete.com/posts/couldnt-irgen-expression/.

It offers a few workarounds but they have tradeoffs. Still, you might find them useful while we work out a better solution. One thing I did notice (that isn't in the article) is that everything 'just works' in a mixed Swift/ObjC project. My guess is that if needed you can unblock yourself by including an empty ObjC file.

Also this is happening because the `FacebookSDK` CocoaPod uses prebuilt XCFrameworks. If you use the individual pods such as `pod 'FBSDKCoreKit'` you won't encounter this issue.

**Comment by joesus** [comment]
@KemalPajevic please delete the Pod directory, clear derived data, and reinstall the pods... this will help us rule out some possible causes.

**Comment by joesus** [comment]
Setting the `SWIFT_SERIALIZE_DEBUGGING_OPTIONS` argument to `NO` appears to have fixed the issue. Will build with this flag for the next release. Thanks for alerting us to this issue!

**Comment by dflems** [comment]
@joesus Looks like the release after your last comment was 9.1.0, but we're having the same issue with that release. I don't see `SWIFT_SERIALIZE_DEBUGGING_OPTIONS` used anywhere in the repo. Was this flag used to produce the binaries for 9.1.0 and later?

**Comment by aakarsh-sasi** [comment]
This issue still persists when using sdk binarybut the path in the issue has changed:
`error: virtual filesystem overlay file '/Users/distiller/Library/Developer/Xcode/DerivedData/FacebookSDK-chdtjnyfsloviifwexeuwgjiyqce/Build/Intermediates.noindex/ArchiveIntermediates/FBSDKCoreKit-Dynamic/IntermediateBuildFilesPath/FBSDKCoreKit.build/Release-iphoneos/FBSDKCoreKit-Dynamic.build/all-product-headers.yaml' not found`

Using --no-use-binaries while building using Carthage solved the issue for me but it takes 30-40 mins fo building all dependencies.

@dflems how do you build with SWIFT_SERIALIZE_DEBUGGING_OPTIONS=NO ? Have to pass this as a flag in xcodebuild?

**Comment by dflems** [comment]
@aakarsh-sasi Not using Carthage. I was just downloading and directly integrating the prebuilt static frameworks. I added the flag to these `xcodebuild` invocations:

https://github.com/facebook/facebook-ios-sdk/blob/48c0203d6a4502bf369ee2171e9ef5d63a3a485c/scripts/xcode/build-universal-framework.sh#L38-L52

Then I ran the universal aggregate targets like:

```
xcodebuild -configuration Release -project FBSDKLoginKit/FBSDKLoginKit.xcodeproj -scheme FBSDKLoginKit-Universal
```

It writes the static frameworks out to the `build` directory

**Comment by aakarsh-sasi** [comment]
@dflems Thanks this is helpful

**Comment by joesus** [comment]
Sounds like that PR should fix this. In the meantime I'd like to take a moment to promote using Swift Package Manager or the XCFrameworks attached to the release. Both options are much easier to integrate. Thanks again for bringing this to our attention!

**Comment by joesus** [comment]
Also locking because the issue relates to 9.0.1 and we are on 9.2.0. Please don't hesitate to open a new issue with details about any regressions. Thanks!
